Adopt more agressive removal policy for alpine and fedora

rvagg opened the pull request:
The previous policy we've adopted is that we'd keep test machines around as long as they don't cause too much maintenance overhead and we have the resources.

I'd like to propose we adjust that for the Linuxes without long-term support lifecycles, in particular Alpine and Fedora, and adopt the policy of simply removing them when they expire support. There's a bunch of reasons but the primary is actually maintenance overhead, but not in the particular that they are a problem, but in the general that we have too many hosts to manage and the more we have in our CI mix the more work we have to do and the more chance of false positive failures.

This PR also remove alpine from vanilla ansible setup--using Docker for alpine now.

gibfahn commented:
>There's a bunch of reasons but the primary is actually maintenance overhead, but not in the particular that they are a problem, but in the general that we have too many hosts to manage and the more we have in our CI mix the more work we have to do and the more chance of false positive failures.

I think this is a reasonable definition of maintenance overhead, and thus completely in keeping with our existing policy.

rvagg commented:
I did this today. Although I left Alpine 3.6 along with 3.7. I looked at the "fedora-latest-x64" and "fedora-last-latest-x64" strategy and decided I really liked it and applied it to Alpine too. I've gone through the Jenkins jobs and removed direct references to Fedora or Alpine versions and used these labels instead. It just means keeping them updated! I discovered that I didn't roll those labels forward when I introduced Fedora 27 so "fedora-latest-x64" was Fedora 26 and "fedora-last-latest-x64" was Fedora 25. Fixed now and I shut down All of the old Fedora machines.

I have not deleted the old Alpine Docker containers from the Docker hosts (that's not as simple as with IaaS deletion), they'll just become static until the next time I deploy over the top of those machines with a new config. I also just introduced a `--without-intl` builder as per #419 so I need to rebalance those Docker hosts with more containers that can serve node-test-commit-linux-containered instead of leaving space for the unused Alpine.

I'm still considering how best to attack documenting all of this, there are a few issues and PRs open relating to policy and documenting these things and I want to pull them all together somehow without making too much work that won't get done.
